src/Raven.Server/Documents/Indexes/Workers/HandleReferences.cs

```diff
@@ -99,8 +99,8 @@ protected HandleReferencesBase(Index index, DocumentsStorage documentsStorage, I
                             ? _configuration.MaxTimeForDocumentTransactionToRemainOpen.AsTimeSpan
                             : TimeSpan.FromMinutes(15);
 
-            var moreWorkFound = HandleItems(ActionType.Tombstone, queryContext, indexContext, writeOperation, stats, pageSize, maxTimeForDocumentTransactionToRemainOpen, token);
-            moreWorkFound |= HandleItems(ActionType.Document, queryContext, indexContext, writeOperation, stats, pageSize, maxTimeForDocumentTransactionToRemainOpen, token);
+            var moreWorkFound = HandleItems(LastProcessedReference.ActionType.Tombstone, queryContext, indexContext, writeOperation, stats, pageSize, maxTimeForDocumentTransactionToRemainOpen, token);
```

Comment:
why nesting that inside LastProcessedReference? It should stay in HandleReferences?

Reply:
done